src/java/org/apache/cassandra/service/accord/serializers/RecoverySerializers.java

```diff
@@ -46,42 +46,56 @@
     {
         @Override
         public void serializeBody(BeginRecovery recover, DataOutputPlus out, Version version) throws IOException
         {
             CommandSerializers.partialTxn.serialize(recover.partialTxn, out, version);
+            int flags =   (recover.route != null ? HAS_ROUTE : 0)
+                        | (recover.executeAtOrTxnIdEpoch != recover.txnId.epoch() ? HAS_EXECUTE_AT_EPOCH : 0)
+                        | (recover.isFastPathDecided ? IS_FAST_PATH_DECIDED : 0);
             CommandSerializers.ballot.serialize(recover.ballot, out);
-            serializeNullable(recover.route, out, KeySerializers.fullRoute);
-            out.writeUnsignedVInt(recover.executeAtOrTxnIdEpoch - recover.txnId.epoch());
+            out.writeUnsignedVInt32(flags);
+            if (recover.route != null)
+                KeySerializers.fullRoute.serialize(recover.route, out);
+            if (0 != (flags & HAS_EXECUTE_AT_EPOCH))
+                out.writeUnsignedVInt(recover.executeAtOrTxnIdEpoch - recover.txnId.epoch());
         }
 
         @Override
         public BeginRecovery deserializeBody(DataInputPlus in, Version version, TxnId txnId, Route<?> scope, long waitForEpoch, long minEpoch) throws IOException
         {
             PartialTxn partialTxn = CommandSerializers.partialTxn.deserialize(in, version);
             Ballot ballot = CommandSerializers.ballot.deserialize(in);
-            @Nullable FullRoute<?> route = deserializeNullable(in, KeySerializers.fullRoute);
-            long executeAtOrTxnIdEpoch = in.readUnsignedVInt32() + txnId.epoch();
-            return BeginRecovery.SerializationSupport.create(txnId, scope, waitForEpoch, minEpoch, partialTxn, ballot, route, executeAtOrTxnIdEpoch);
+            int flags = in.readUnsignedVInt32();
+            FullRoute<?> route = null;
+            if (0 != (flags & HAS_ROUTE))
+                route = KeySerializers.fullRoute.deserialize(in);
+            long executeAtOrTxnIdEpoch = txnId.epoch();
+            if (0 != (flags & HAS_EXECUTE_AT_EPOCH))
+                executeAtOrTxnIdEpoch += in.readUnsignedVInt32();
+            boolean isFastPathDecided = 0 != (flags & IS_FAST_PATH_DECIDED);
+            return BeginRecovery.SerializationSupport.create(txnId, scope, waitForEpoch, minEpoch, partialTxn, ballot, route, executeAtOrTxnIdEpoch, isFastPathDecided);
         }
 
         @Override
         public long serializedBodySize(BeginRecovery recover, Version version)
         {
             return CommandSerializers.partialTxn.serializedSize(recover.partialTxn, version)
                    + CommandSerializers.ballot.serializedSize(recover.ballot)
-                   + serializedNullableSize(recover.route, KeySerializers.fullRoute)
-                   + TypeSizes.sizeofUnsignedVInt(recover.executeAtOrTxnIdEpoch - recover.txnId.epoch());
+                   + VIntCoding.computeUnsignedVIntSize(IS_FAST_PATH_DECIDED)
```

Comment:
Should we leave a comment somewhere to indicate fast path is _currently_ the highest so we don't miss this in future?